appdatabase/database.go

```diff
@@ -198,11 +188,144 @@ func MigrateEnsUsernames(db *sql.DB) error {
 			continue
 		}
 
-		_, err = db.Exec(`INSERT INTO ens_usernames (username, chain_id) VALUES (?, ?)`, username, defaultChainID)
+		_, err = sqlTx.Exec(`INSERT INTO ens_usernames (username, chain_id) VALUES (?, ?)`, username, defaultChainID)
 		if err != nil {
 			log.Error("Migrating ens usernames: failed to insert username into new database", "ensUsername", username, "err", err.Error())
 		}
 	}
 
 	return nil
 }
+
+const (
+	batchSize = 1000
+)
+
+func migrateWalletJSONBlobs(sqlTx *sql.Tx) error {
+	var batchEntries [][]interface{}
+
+	// Extract useful information from the receipt blob and store it as sql interpretable
+	//
+	// Dropped storing postState because it was replaced by the status after EIP 658
+	// Dropped duplicating logs until we have a better way to store them in a
```

Comment:
The `log` entry points to the log that originated the ERC20/ERC721 transfer. Agree about not duplicating, but if we do not store a copy of the log, we should at least store its index (`view.go` uses it to get the actual information we need about the transfer)

Reply:
> The log entry points to the log that originated the ERC20/ERC721 transfer. Agree about not duplicating, but if we do not store a copy of the log, we should at least store its index (view.go uses it to get the actual information we need about the transfer

Indeed log entry is duplicated; however, it makes sense to migrate the index now. WDYT?

> IMO this kind of metadata should be stored in a DB -multitransaction?- and not generated on the fly)

We discussed refactoring the entire data; however, given the time constraints, I decided that this would be too much work that won't benefit the short-term activity feature and leave it for some further reorganization based on the new requirements later on.

Regarding why no multi-transaction for the new fields is because not all transactions in transfers are recorded in multi-transaction and for the current work there is not much difference.

Reply:
With the latest changes I migrated also the `log_index` by extracting it from the log column.